pkg/util/deployment/deployment.go

```diff
@@ -140,11 +140,28 @@ func ListPods(deployment *extensions.Deployment, getPodList podListFunc) (*api.P
 	return getPodList(namespace, options)
 }
 
+// equalIgnoreHash returns true if two given podTemplateSpec are equal, ignoring the diff in value of Labels[pod-template-hash]
+// We ignore pod-template-hash because the hash result would be different whenever podTemplateSpec API changes
+func equalIgnoreHash(rsTemplate, template api.PodTemplateSpec) bool {
+	if len(rsTemplate.Labels) == 0 && len(template.Labels) == 0 || rsTemplate.Labels[extensions.DefaultDeploymentUniqueLabelKey] == template.Labels[extensions.DefaultDeploymentUniqueLabelKey] {
+		return api.Semantic.DeepEqual(rsTemplate, template)
+	}
+	// We expect them to both have/not have pod-template-hash label
+	if len(rsTemplate.Labels) == 0 || len(template.Labels) == 0 || len(rsTemplate.Labels[extensions.DefaultDeploymentUniqueLabelKey]) == 0 || len(template.Labels[extensions.DefaultDeploymentUniqueLabelKey]) == 0 {
```

Comment:
Is it possible for ` len(rsTemplate.Labels) == 0 || len(template.Labels) == 0` to ever be false?